### DISCUSSION TITLE
How do I run open a process in a new tab?

### DISCUSSION THREAD

**Discussion by ayan2k26** [wezterm/wezterm#3451] [Q&A] [answered]
I am using ranger as my file manager and ranger can open files directly and if the app to open it is a terminal based one, like say nvim, then it opens it in the same window and upon closing the window it goes back to the ranger screen. What I want to do is open a file from ranger in a new tab. From what I have gathered, it is a terminal based options which sends the command to a new window.

The default options to open a text file in ranger is defined in the config file as follows -
```
mime ^text, label editor = ${VISUAL:-$EDITOR} -- "$@"
```
which can be changed to the following for termite (from https://superuser.com/questions/1388604/ranger-open-with-vim-in-new-windo)
```
mime ^text,  label editor = nohup termite -e "${VISUAL:-$EDITOR} $@" >/dev/null &
```

How can I achieve the same in wezterm?

**Comment by wez** [comment]
Take a look at: https://wezfurlong.org/wezterm/cli/cli/spawn.html

**Reply by ayan2k26** [reply, marked as answer]
Thanks @wez, I seem to have gotten it to work.
Worked - `mime ^text,  label editor = wezterm cli spawn -- nvim -- "$@"`
Didn't work - `mime ^text,  label editor = wezterm cli spawn -- ${VISUAL:-$EDITOR} -- "$@"`

**Reply by wez** [reply]
maybe try running a shell to evaluate the environment:

`mime ^text,  label editor = wezterm cli spawn -- sh -c 'exec ${VISUAL:-$EDITOR} -- "$@"'`

**Reply by ayan2k26** [reply]
This opened nvim in a new tab but not the file, and returned the id of the spawned tab in the terminal from which ranger was called.
